### ISSUE TITLE
Add support for dispersion more generally

### ISSUE THREAD

**Issue by bernstei** [ACEsuit/mace#1535]
It would be helpful, and quite easy, to add dispersion support to `MACECalculator`, same as is currently done in `mace_mp` with `SumCalculator`.  It would also be useful, if somewhat more work, to add dispersion support to training, which would automatically _subtract_ dispersion energies/forces/stresses from the fitting/validation/testing data.

**Comment by bernstei** [comment]
I've thought about this a bit more, and it's not quite as easy as I initially thought, because you can't trivially wedge the `SumCalculator` into the `MACECalculator` constructor.  My thoughts so far are to create the D3 calculator, store it in the `MACECalculator.dispersion_calc`, and add the results manually in `MACECalculator.calculate()`.  

Another option is to rename `MACECalculator` to `MACECalculator_ll`, and a new `MACECalculator` that either returns the raw `MACECalculator_ll` or a `SumCalculator` depending on whether or not dispersion was requested.  But this seems messy as well.

**Comment by jwjeffr** [comment]
I'm actually running into this exact problem, we're trying to fine-tune for some Moire bilayers

This seems to be a pretty specialized case, I think? I've also thought about it a bit for the last couple of weeks, and I'm in strong agreement that it seems like every reasonable solution is going to be messy. But I'd be happy to be wrong 🙂

If I'm not wrong, it might be worth just writing a tutorial transforming a dataset to do this, something like:

- load in some DFT data
- compute D3 correction energies and forces
- store diffs in new keys, something like `dft_minus_d3_energy` and `dft_minus_d3_forces`
- train on new keys
- deploy model + D3 correction on a test set

I'd be happy to write such a tutorial as well

**Comment by gabor1** [comment]
I would like us to make contact with this work: https://arxiv.org/abs/2607.15103 . Previously people were caught out by using naive dispersion implementations for larger systems whose evaluation time far exceeded the mace evaluation time.
